UPGRADING

```diff
@@ -19,6 +19,16 @@ PHP 8.4 UPGRADE NOTES
 1. Backward Incompatible Changes
 ========================================
 
+- Core:
+  . isset() and empty() now check that the offset for string values is of the
+    correct type i.e. numeric, similarly to array and objects.
+    This means code similar to:
+    $a = 'string';
+    var_dump(isset($a['b']));
+    var_dump(empty($a['b']));
```

Comment:
btw: PHPStan also warns in this constellation:

https://phpstan.org/r/3b26cd4f-bdd1-470a-9b8f-1f42eb1386ee